DOC-2334 doc(cdc): [3.10] cdc setup doc improvement; (#674)
tigergraph/server-docs: modules/system-management/pages/change-data-capture/cdc-setup.adoc

gadmin config set System.CDC.ProducerConfig @/home/tigergraph/test_cdc/cdc_producer_config
----
The full list of entries for configuration file of `System.CDC.ProducerConfig` can be viewed in table "Global configuration properties" here: https://github.com/confluentinc/librdkafka/blob/v1.1.0/CONFIGURATION.md#global-configuration-properties.

Only properties marked with P(Producer) or *(both Producer and Consumer) are applicable. 

==== Kafka Security

> There are some entries of `System.CDC.ProducerConfig` to enhance security of communication to external kafka, besides the must-have entry `bootstrap.servers`.

Reviewer: ```suggestion
To enhance the security of communication with external Kafka, you can configure certain entries in the `System.CDC.ProducerConfig`. In addition to the mandatory `bootstrap.servers` entry, other configuration properties can help secure the communication.

The complete list of configuration entries for `System.CDC.ProducerConfig` is available in the Global Configuration Properties table in the [librdkafka documentation](https://github.com/confluentinc/librdkafka/blob/v1.1.0/CONFIGURATION.md#global-configuration-properties).

Only the properties marked with P (Producer) or * (both Producer and Consumer) are applicable for this purpose. You can reference these properties to ensure your Kafka setup adheres to security best practices.
```

Reviewer 2: @anjaliIthapeTG Your lines 78-80 are repeating lines 70-72 above. Are you proposing to repeat them or move them?

[NOTE]
====

> 1. When using local path in any entry, for example: `sasl.kerberos.keytab` or `ssl.ca.location`, the local file must exist and be consistent on all nodes in TigerGraph cluster.

Reviewer: ```suggestion
1. When using local file paths in any entry (e.g., `sasl.kerberos.keytab` or`ssl.ca.location`), the local file must exist and be consistent across all nodes in the TigerGraph cluster.```

> 2. The entry `sasl.jaas.config` is not applicable, because it is only for JAVA-based kafka client, while librdkafka in TigerGraph engine is C++ library.

Reviewer: ```suggestion
2. The entry `sasl.jaas.config` is not applicable because it is specific to Java-based Kafka clients, while TigerGraph engine uses the librdkafka C++ library, so this property is unsupported.
```

====

Example 1: Authenticating with SASL/PLAIN

[console]
----
security.protocol=SASL_PLAINTEXT
sasl.mechanisms=PLAIN
sasl.username=<username>
sasl.password=<password>
----

Example 2: Authenticating with SASL/GSSAPI
[console]
----
security.protocol=SASL_PLAINTEXT
sasl.mechanism=GSSAPI
sasl.kerberos.service.name=kafka
sasl.kerberos.principal=<user@EXAMPLE.COM>
sasl.kerberos.keytab=</path/to/user.keytab>
----

> More reference: https://github.com/confluentinc/librdkafka/wiki/Using-SASL-with-librdkafka

Reviewer 2: ```suggestion
Additional reference: https://github.com/confluentinc/librdkafka/wiki/Using-SASL-with-librdkafka
```

Example 3: Authenticating with SASL/PLAIN and Encrypted with SSL
[console]
----
security.protocol=SASL_SSL
sasl.mechanisms=PLAIN
sasl.username=<username>
sasl.password=<password>
ssl.ca.location=<path/to/ca.pem>
ssl.certificate.location=<path/to/cert.pem>
ssl.key.location=<path/to/key.pem>
----

> 

Reviewer: ```suggestion
Additional Resources
For more information on using SASL with librdkafka, refer to https://github.com/confluentinc/librdkafka/wiki/Using-SASL-with-librdkafka.
```
